src/main/java/org/generationcp/middleware/service/impl/dataset/DatasetServiceImpl.java

```diff
@@ -1121,6 +1108,54 @@ public void setValueToVariable(final Integer datasetId, final ObservationUnitsPa
 		}
 	}
 
+	@Override
+	public void deleteVariableValues(final Integer studyId, final Integer datasetId, final ObservationUnitsSearchDTO searchDTO) {
+		final String variableId = searchDTO.getFilter().getVariableId().toString();
+		final Boolean draftMode = searchDTO.getDraftMode();
+
+		final List<ObservationUnitRow> observationUnitsByVariable =
+			this.getObservationUnitsByVariable(datasetId, searchDTO, studyId);
+
+		final List<Integer> phenotypeIdsToDelete = new ArrayList<>();
+		final Set<Integer> observationUnitIdsToUpdate = new HashSet<>();
+		if (!observationUnitsByVariable.isEmpty()) {
+			for (final ObservationUnitRow observationUnitRow : observationUnitsByVariable) {
+				final ObservationUnitData observationUnitData = observationUnitRow.getVariables().get(variableId);
+				Phenotype phenotype = null;
+
+				if (observationUnitData != null) {
+					phenotype = this.daoFactory.getPhenotypeDAO().getById(observationUnitData.getObservationId());
+				}
+
+				if (phenotype != null) {
+					if (draftMode) {
+						if(phenotype.getcValueId() == null && StringUtils.isEmpty(phenotype.getValue())) {
+							phenotypeIdsToDelete.add(phenotype.getPhenotypeId());
+						} else {
+							this.updatePhenotype(phenotype,
```

Comment:
@mderamos seems to me that it is possible to run only one update instead of calling updates n-times